resilience4j-timelimiter/src/main/java/io/github/resilience4j/timelimiter/internal/TimeLimiterImpl.java

```diff
@@ -48,11 +48,12 @@ public TimeLimiterImpl(String name, TimeLimiterConfig timeLimiterConfig,
                 onSuccess();
                 return result;
             } catch (TimeoutException e) {
-                onError(e);
+                TimeoutException timeoutException = createdTimeoutExceptionWithName(name);
```

Comment:
Should we copy the source stacktrace of the original TimeoutException?

Reply:
Good point. 
Just adjusted it.